test/e2e/snaps/ppom-toggle-settings.spec.js

```diff
@@ -0,0 +1,87 @@
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  unlockWallet,
+  openDapp,
+  defaultGanacheOptions,
+  getWindowHandles,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+const mainnetProviderConfig = {
+  providerConfig: {
+    chainId: '0x1',
+    nickname: '',
+    rpcUrl: '',
+    type: 'mainnet',
+  },
+};
+
+describe('PPOM Settings', function () {
+  it('should not show the PPOM warning when toggle is off', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkController(mainnetProviderConfig)
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        await openDapp(driver);
+        await driver.clickElement('#maliciousPermit');
+        const windowHandles = await getWindowHandles(driver, 3);
+        await driver.switchToWindow(windowHandles.popup);
+
+        const blockaidResponseTitle =
+          '[data-testid="security-provider-banner-alert"]';
+        const exists = await driver.isElementPresent(blockaidResponseTitle);
+        assert.equal(exists, false, 'This is a deceptive request');
```

Comment:
it looks like this is attempting to test title
> assert.equal(exists, true, 'This is a deceptive request');

however, the third parameter is meant to override the error message. In the test, the actual warning provided shows "Request may not be safe", not "This is deceptive request". The assert above only checks that it exists

from /node_modules/@types/node/assert.d.ts:
```
         * If the values are not equal, an `AssertionError` is thrown with a `message`property set equal to the value of the `message` parameter. If the `message`parameter is undefined, a default
         * error message is assigned. If the `message`parameter is an instance of an `Error` then it will be thrown instead of the`AssertionError`.
         * @since v0.1.21
         */
        function equal(actual: unknown, expected: unknown, message?: string | Error): void;
```

we should update this in a future PR